client/src/shared/state/Inventory.js

```diff
@@ -104,16 +110,76 @@ class Inventory {
         if (!ItemTypes[itemConfig.typeCode].hasUseEffect) return;
 
         this.hotbar.push(itemConfig);
-
+        this.saveItemId[itemConfig.id] = itemConfig.slotIndex;
         PubSub.publish(HOTBAR_ITEM);
         PubSub.publish(MODIFY_INVENTORY_ITEM);
+        if (this.saveSession === true) {
+            this.saveHotbar();
+        }
     }
 
     removeFromHotbar(itemConfig) {
+        delete this.saveItemId[itemConfig.id];
         this.hotbar = this.hotbar.filter((eachItem) => eachItem !== itemConfig);
-
         PubSub.publish(HOTBAR_ITEM);
         PubSub.publish(MODIFY_INVENTORY_ITEM);
+        if (this.saveSession === true) {
+            this.saveHotbar();
+        }
+    }
+
+    saveHotbar() {
+        window.localStorage.setItem(this.loadHotBarRequest, JSON.stringify(this.saveItemId));
+    }
+
+    defaultHotBar() {
+        this.items.forEach((itemConfig) => {
+            this.addToHotbar(itemConfig);
+        });
+    }
+
+    loadFromLocalStorage(playerID) {
+        this.saveSession = true;
+        this.userName = playerID;
+        const gameName = "dungeonz";
+        const requestHotbar = "hotBar";
+        this.loadHotBarRequest = gameName + this.userName + requestHotbar;
+    }
+
+    loadHotbar(playerID, isLoggedIn) {
+        if (isLoggedIn === true) {
+            this.loadFromLocalStorage(playerID);
+        }
+        const loadStorageHotbar = window.localStorage.getItem(this.loadHotBarRequest);
+        if (this.saveSession === false || loadStorageHotbar === null) {
+            this.defaultHotBar();
+            return;
+        }
+        this.saveItemId = JSON.parse(loadStorageHotbar);
+        this.populateInvetorytoHotbar();
+    }
+
+    populateInvetorytoHotbar() {
+        Object.keys(this.saveItemId).forEach((key, value) => {
+            this.items.forEach((itemConfig) => {
+                if (key === itemConfig.id) {
+                    this.addToHotbar(itemConfig);
+                }
+            });
+        });
+
+        // For the case there is a mismatch in session remove non-existing keys
+        Object.keys(this.saveItemId).forEach((key, value) => {
+            this.foundKey = false;
```

Comment:
I think `foundKey` can be scoped to this callback function instead of being on the state object as it isn't used outside of the function.
i.e.
`let foundKey = false;`